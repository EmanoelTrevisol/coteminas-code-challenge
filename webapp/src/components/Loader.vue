<template>
  <div class="loader">
    <font-awesome-icon icon="spinner" spin size="4x"></font-awesome-icon>
  </div>
</template>

<script>
export default {};
</script>

<style lang="stylus" scoped>
.loader {
	height: 40vh;
	display: flex;
	justify-content: center;
	align-items: center;
}
</style>
