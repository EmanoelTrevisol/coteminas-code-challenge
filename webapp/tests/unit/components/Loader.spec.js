import { shallowMount } from '@vue/test-utils';
import Loader from '@/components/Loader';

describe('Loader.vue', () => {
  it('renders Loader component', () => {
    const wrapper = shallowMount(Loader);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('passes correct icon, spin directive and size to FontAwesomeIcon', () => {
    const wrapper = shallowMount(Loader);

    const icon = wrapper.find('font-awesome-icon');

    expect(icon.exists()).toBe(true);
    expect(icon.attributes('icon')).toBe('spinner');
    expect(icon.attributes('spin')).toBeDefined();
    expect(icon.attributes('size')).toBe('4x');
  });
});
