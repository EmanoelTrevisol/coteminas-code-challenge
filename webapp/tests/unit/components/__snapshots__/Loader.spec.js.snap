// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader.vue renders Loader component 1`] = `
<div class="loader">
  <font-awesome-icon icon="spinner" spin="" size="4x"></font-awesome-icon>
</div>
`;
